### PR TITLE
Update cantera to version 3.0.0

### PR DESCRIPTION
**Pull request by tulioricci**
- ~Depends on  https://github.com/pyrometheus/pyrometheus/pull/75~
- Closes https://github.com/illinois-ceesd/mirgecom/issues/988
- Points to a new pyrometheus branch streamlined to be merged to main
    - https://github.com/pyrometheus/pyrometheus/pull/79 
- Remove spatial dependence in `autoignition.py` and further modify it to easy the comparison against pyrometheus and cantera
- Search mechanism in the local working folder if it is not in the default `mechanism` location

**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: test/test_eos.py [hunk 1]
```diff
@@ -302,7 +300,7 @@ def inf_norm(x):
         pyro_t = pyro_mechanism.get_temperature(pyro_e, tin, yin)
         pyro_p = pyro_mechanism.get_pressure(pyro_m, pyro_t, yin)
         pyro_h = pyro_mechanism.get_mixture_enthalpy_mass(pyro_t, yin)
-        pyro_s = pyro_mechanism.get_mixture_entropy_mass(pyro_p, pyro_t, yin)
+        # pyro_s = pyro_mechanism.get_mixture_entropy_mass(pyro_p, pyro_t, yin)
```
Comment (MTCam) [inline test/test_eos.py:303]: OOC, how's come?

File: examples/autoignition.py [hunk 2]
```diff
@@ -164,13 +185,6 @@ def main(actx_class, use_leap=False, use_overintegration=False,
     nodes = actx.thaw(dcoll.nodes())
     ones = dcoll.zeros(actx) + 1.0
 
-    if use_overintegration:
-        quadrature_tag = DISCR_TAG_QUAD
-    else:
-        quadrature_tag = None
```
Comment (MTCam) [inline examples/autoignition.py:170]: Why remove?

**Review by MTCam — COMMENTED** [review]
I am not sure I am comfortable removing the fluid stuff for this example.   It is not wasted on me that it is largely non-participant in this example, but part of the example was showing how to have chem+fluid in the simulation.

File: test/test_eos.py (hunk 1 above)
Comment (tulioricci) [reply test/test_eos.py:303]: Per Pyrometheus review, we removed entropy from its native functions. I am wondering if I should remove it completely or add local functions somewhere to keep the verification.

Comment (MTCam) [reply test/test_eos.py:303]: If it is no longer in the base package and the corresponding test was removed there, then this is the right thing to do. I was just curious about the removal.

**Comment by tulioricci** [comment]
> I am not sure I am comfortable removing the fluid stuff for this example. It is not wasted on me that it is largely non-participant in this example, but part of the example was showing how to have chem+fluid in the simulation.

My plan is to make `autoignition` for chemistry only (faster execution) and to add a 1D flame driver to verify spatial dependence + mixture transport models. I've been using both these cases for verification/debugging of chemistry in mirge-com. To avoid excess/superfluous examples, I want to remove `mixture.py`. See https://github.com/illinois-ceesd/mirgecom/pull/995

However, I can keep the spatial terms in `autoignition`. No big deal.

**Comment by MTCam** [comment]
> My plan is to make `autoignition` for chemistry only (faster execution) and to add a 1D flame driver to verify spatial dependence + mixture transport models. I've been using both these cases for verification/debugging of chemistry in mirge-com. To avoid excess/superfluous examples, I want to remove `mixture.py`. See #995
> 
> However, I can keep the spatial terms in `autoignition`. No big deal.

I'm not opposed.  Hearing your plan makes it seem like a good idea.

File: test/test_eos.py (hunk 1 above)
Comment (tulioricci) [reply test/test_eos.py]: Removed. :+1:

**Review by MTCam — APPROVED** [review]

**Pull request merged**